include/calculate.h

```diff
@@ -28,42 +28,47 @@ namespace calculate {
 
 
     class Calculate final {
+        String _expression;
+        vString _variables;
         pValue _values;
         pEvaluable _tree;
 
-        qSymbol _tokenize(const String &expression) const;
-        qSymbol _check(qSymbol &&input) const;
-        qEvaluable _shuntingYard(qSymbol &&infix) const;
-        pEvaluable _buildTree(qEvaluable &&postfix) const;
+        static qSymbol _tokenize(const String &expr, const vString &vars,
+                                 const pValue &values);
+        static qSymbol _check(qSymbol &&input);
+        static qEvaluable _shuntingYard(qSymbol &&infix);
+        static pEvaluable _buildTree(qEvaluable &&postfix);
 
         static vString _extract(const String &vars);
         static vString _validate(const vString &vars);
 
         Calculate() = delete;
-        Calculate& operator=(const Calculate &other) = delete;
-        Calculate& operator=(Calculate &&other) = delete;
 
     public:
-        const String expression;
-        const vString variables;
-
         Calculate(const Calculate &other);
         Calculate(Calculate &&other);
         Calculate(const String &expr, const String &vars);
         Calculate(const String &expr, const vString &vars={});
 
+        Calculate& operator=(const Calculate &other);
+        Calculate& operator=(Calculate &&other);
+
         bool operator==(const Calculate &other) const noexcept;
+
         double operator() () const;
         double operator() (double value) const;
         double operator() (vValue values) const;
 
         template <typename Head, typename... Tail>
         double operator() (Head head, Tail... tail) const {
-            if (sizeof...(tail) + 1 > variables.size())
+            if (sizeof...(tail) + 1 > _variables.size())
                 throw WrongArgumentsException();
-            _values[variables.size() - sizeof...(tail) - 1] = head;
+            _values[_variables.size() - sizeof...(tail) - 1] = head;
             return this->operator() (tail...);
         };
+        
+        const String& getExpression() const noexcept {return _expression;}
+        const vString& getVariables() const noexcept {return _variables;}
```

Comment:
Constant references were chosen as the return value to been able to use the `c_str` function on the returned expression ([read more](http://stackoverflow.com/a/5095204)).